modules/client/src/main/java/org/apache/ignite/internal/client/TcpClientChannel.java

```diff
@@ -782,4 +803,120 @@ private class HeartbeatTask extends TimerTask {
             }
         }
     }
+
+    /**
+     * Timeout object worker.
+     */
+    private static class TimeoutWorker extends IgniteWorker {
```

Comment:
`TimeoutWorker` classes seem almost identical and `TimeoutObject` classes only differ in the element type of the future. It seems that it's really easy to have just one generic `TimeoutObject`. As for `TimeoutWorker`, most its logic is identical, and differences can be extracted to an abstract method, to avoid code duplication.

Reply:
I moved this class to another place, where it is available for both usages.